File: python palindrome/palindrome.py
#-
# Example palindrome program
#
# Run from command line with 1 argument ->
# python ./palindrome.py pal.txt
#-

import sys
import string

def sarah_palindrome ():
	cries = get_text()
	map (checkpalindrome,cries)

def get_text ():
# Read file passed from command line arg

	textfile_path = sys.argv[1]
	textfile = open(textfile_path,'r')
	text = list(textfile)
	return [x.strip() for x in text]

def checkpalindrome ( word ):
# Checks if the string is a palindrome

	if (word.lower() == word[::-1].lower()):
		print 'AY |', r_lexical(word)
	else:
		print 'NAY |', r_lexical(word)

def r_lexical ( word ):
# Sort and format for output

	word = word.replace(' ','')
	li = list(word)
	li.sort()
	li.sort(key=lambda x:(x.islower(), x))
	s = ''.join(li)
	return s[::-1]

sarah_palindrome()

File: rust palindrome/palindrome.rs
//-
// Example palindrome program

// Compile first ->
// rustc nazgul.rs

// Run from command line with 1 argument ->
// ./nazgul pal.txt
//-

use std::env;
use std::fs::File;
use std::io::{Read,BufReader};


fn main() {
// Read file passed from command line arg

	if let Some(arg1) = env::args().nth(1) {
		let mut data = String::new();
	    let f = File::open(arg1).unwrap();
	    let mut br = BufReader::new(f);
	    br.read_to_string(&mut data).unwrap();
	    for line in data.lines() {
	    	checkpalindrome(line);
	    }
	}
}

fn checkpalindrome(s: &str) {
// Check if str in palindrome

	let s_rev = s.chars().rev().collect::<String>();
	if s.to_lowercase() == s_rev.to_lowercase() {
		r_lexical("AY |",s);
	} else {
		r_lexical("NAY |",s);
	}
}

fn r_lexical(p: &str, s: &str) {
// Sort and format for output

    let mut chars: Vec<char> = s.chars().collect();
    chars.sort_by(|first, second| second.cmp(first));
    let mut clean = String::new();
    for c in chars {
    	clean.push(c);
    }
    clean.replace(" ","");

    println!("{0} {1}", p, clean);
}